test/libethereum/blockchain.cpp

```diff
@@ -124,632 +85,425 @@ void doBlockchainTests(json_spirit::mValue& _v, bool _fillin)
 
-				// write block header
-				mObject oBlockHeader;
-				writeBlockHeaderToJson(oBlockHeader, current_BlockHeader);
-				blObj["blockHeader"] = oBlockHeader;
-				vBiBlocks.push_back(current_BlockHeader);
-
-				// compare blocks from state and from rlp
-				RLPStream txStream;
-				txStream.appendList(txList.size());
-				for (unsigned i = 0; i < txList.size(); ++i)
+				bool wasException = false;
+				compareBlocks(block, alterBlock);
+				try
 				{
-					RLPStream txrlp;
-					txList[i].streamRLP(txrlp);
-					txStream.appendRaw(txrlp.out());
+					blockchain.addBlock(alterBlock);
+					trueBc.addBlock(alterBlock);
+					if ((string)boost::unit_test::framework::current_test_case().p_name != "bcWalletTest")
+						importedBlocks.push_back(alterBlock);
 				}
-
-				RLPStream blockRLP = createFullBlockFromHeader(current_BlockHeader, txStream.out(), uncleStream.out());
-
-				blObj["rlp"] = toHex(blockRLP.out(), 2, HexPrefix::Add);
-
-				if (sha3(RLP(block.blockData())[0].data()) != sha3(RLP(blockRLP.out())[0].data()))
-				{				
-					cnote << "block header mismatch block.blockData() vs updated block.info()\n";
-					cnote << toHex(RLP(block.blockData())[0].data()) << "vs" << toHex(RLP(blockRLP.out())[0].data());
+				catch (Exception const& _e)
+				{
+					cnote << testname + "block import throw an exception: " << diagnostic_information(_e);
+					wasException = true;
 				}
-
-				if (sha3(RLP(block.blockData())[1].data()) != sha3(RLP(blockRLP.out())[1].data()))
-					cnote << "txs mismatch\n";
-
-				if (sha3(RLP(block.blockData())[2].data()) != sha3(RLP(blockRLP.out())[2].data()))
-					cnote << "uncle list mismatch\n" << RLP(block.blockData())[2].data() << "\n" << RLP(blockRLP.out())[2].data();
-
-				try
+				catch (std::exception const& _e)
 				{
-					block.sync(bc);
-					bc.import(blockRLP.out(), block.db());
-					block.sync(bc);
-
-					//there we get new blockchain status in state which could have more difficulty than we have in trueState
-					//attempt to import new block to the true blockchain
-					trueBc.sync(uncleBlockQueue, trueState.db(), 4);
-					trueBc.attemptImport(blockRLP.out(), trueState.db());
-
-					if (block.blockData() == trueBc.block())
-						trueState = block.state();
-
-					blockSet newBlock;
-					newBlock.first = blockRLP.out();
-					newBlock.second = uncleBlockQueueList;
-					if (importBlockNumber < blockSets.size())
-					{
-						//make new correct history of imported blocks
-						blockSets[importBlockNumber] = newBlock;
-						for (size_t i = importBlockNumber + 1; i < blockSets.size(); i++)
-							blockSets.pop_back();
-					}
-					else
-						blockSets.push_back(newBlock);
+					cnote << testname + "block import throw an exception: " << _e.what();
+					wasException = true;
 				}
-				// if exception is thrown, RLP is invalid and no blockHeader, Transaction list, or Uncle list should be given
 				catch (...)
 				{
+					wasException = true;
+				}
+
+				if (wasException)
```

Comment:
instead of using the `wasException` bool, its nicer to write a little helper function and call that in the `catch` block